core/src/test/scala/integration/kafka/admin/TopicCommandIntegrationTest.scala

```diff
@@ -733,12 +733,18 @@ class TopicCommandIntegrationTest extends KafkaServerTestHarness with Logging wi
       killBroker(0)
       val aliveServers = brokers.filterNot(_.config.brokerId == 0)
       TestUtils.waitForPartitionMetadata(aliveServers, underMinIsrTopic, 0)
-      val output = TestUtils.grabConsoleOutput(
-        topicService.describeTopic(new TopicCommandOptions(Array("--under-min-isr-partitions"))))
+      var output = ""
+      TestUtils.waitUntilTrue(
+        () => {
+          output = TestUtils.grabConsoleOutput(
+            topicService.describeTopic(new TopicCommandOptions(Array("--under-min-isr-partitions"))))
```

Comment:
The solution here makes sense if this is really the cause. however, it's unnecessary to use `topicService.describeTopic` here, we can directly get  brokerServer and MetadataCache here.

Reply:
I think the reason the test is failing is `assertTrue(rows(1).startsWith(s"\tTopic: $offlineTopic"))` here we aren't able to get 2 rows. So, I thought of waiting on the same command which was fetching the rows. Am I missing something?

Reply:
Thanks @dengziming .. An issue I see with this is that we need to also validate the topic names that are returned after grabFromConsoleOutput. I guess that won't be possible with the approach you described above.

Reply:
In fact, MetadataCache contains all information about topic partitions including isr, here we can change it like this:
```
      TestUtils.waitUntilTrue(
        () => aliveServers.forall(broker => broker.metadataCache.getPartitionInfo(underMinIsrTopic, 0).get.isr().size() < 6 && broker.metadataCache.getPartitionInfo(offlineTopic, 0).isEmpty),
        "todo"
      )
```

Reply:
you can refer to #12112 as an example.

Reply:
Ok thanks for the example. I made the changes. Plz review

Reply:
Thank you for the update, even the flakiness is brought by underMinIsrTopic, it's still better to wait for offlineTopic metadata as well here?

Reply:
yeah.. i added it back @dengziming

Reply:
Sorry, it seems we won't remove a broker from isr if it's the only replica so this change doesn't make sense, to make it work as expected we should change it like this:
```
xxxx && broker.metadataCache.getPartitionInfo(offlineTopic, 0).get.leader() == MetadataResponse.NO_LEADER_ID
```

Reply:
@dengziming , looks like the latest build failed with a timeout =>
```
org.opentest4j.AssertionFailedError: Timeout waiting for partition metadata propagating to brokers for underMinIsrTopic topic
```
Sure I will take a look at that..